diff --git a/build/jam/packages/Haiku b/build/jam/packages/Haiku
index 5c896d7273..5d1e52bc05 100644
--- a/build/jam/packages/Haiku
+++ b/build/jam/packages/Haiku
@@ -145,10 +145,8 @@ SEARCH on $(userLaunchScripts) = [ FDirName $(HAIKU_TOP) data launch ] ;
 AddFilesToPackage data user_launch : $(userLaunchScripts) ;
 
 # first login scripts
-local firstLoginFiles = default_deskbar_items.sh ;
-firstLoginFiles = $(firstLoginFiles:G=first-login) ;
-SEARCH on $(firstLoginFiles)
-	= [ FDirName $(HAIKU_TOP) data system boot first_login ] ;
+local firstLoginDir = [ FDirName $(HAIKU_TOP) data system boot first_login ] ;
+local firstLoginFiles = [ Glob $(firstLoginDir) : *.sh ] ;
 AddFilesToPackage boot first-login : $(firstLoginFiles) ;
 
 # artwork and sounds
